x-pack/plugins/enterprise_search/public/applications/app_search/routes.ts

```diff
@@ -29,6 +29,9 @@ export const ENGINE_ANALYTICS_PATH = '/analytics';
 
 export const ENGINE_DOCUMENTS_PATH = '/documents';
 export const ENGINE_DOCUMENT_DETAIL_PATH = `${ENGINE_DOCUMENTS_PATH}/:documentId`;
+export const getDocumentDetailRoute = (engineName: string, documentId: string) => {
```

Comment:
[Totally random aside, not related to your PR] Something I played around with a bit at some point but never committed to code - I wonder if it's worth refactoring at some point to store these route helpers in `EngineLogic` which automatically has access to the `engineName` value - that way we can add route helpers without constantly having to pass `engineName` around

Reply:
Yeah, I guess that would make sense if you're inside an Engine related page and have reference to an engine in EngineLogic. But if you were on a totally unrelated page outside of an Engine and wanted to build a link to an Engine page it wouldn't help you. You'd still need to have the helper available elsewhere.

Reply:
yeah, I was thinking it could be an optional arg thing? You can pass in `engineName` but if you don't, it defaults to whatever is stored in EngineLogic 🤔 

Some day!!